### PR TITLE
Finalise transactions before making a snapshot and before restoring

### PR DESCRIPTION
**Pull request by mjlescano**

### CODE REVIEW

File: protocol/synthetix/test/integration/bootstrap.ts [hunk 1]
```diff
@@ -50,10 +50,29 @@ export function bootstrap() {
...
+  /*
+   * Monkey patch to implicitily call to txRequest.wait() method on all write operations.
+   * This is necessary so we are sure that on each test case we don't leave pending txs.
+   */
+  const signers = () => {
+    const results = getSigners();
+
+    for (const signer of results) {
+      const originalSendTransaction = signer.sendTransaction.bind(signer);
+      signer.sendTransaction = async function sendTransactionWithWait(...params) {
```
Comment (noisekit) [inline protocol/synthetix/test/integration/bootstrap.ts:62]: Haha wow that's one way to fix it.

**Review by noisekit — APPROVED** [review]
Amazing. Send it

**Review by noisekit — APPROVED** [review]
As long as ci green. Send it

File: protocol/synthetix/test/common/stakers.ts [hunk 2]
```diff
@@ -26,9 +27,11 @@ export function bootstrapStakers(
   });
 
   before('create traders', async () => {
-    await stake(systems(), 2, 1000, staker1, delegateAmount);
-    await stake(systems(), 2, 1001, staker2, delegateAmount);
-    await stake(systems(), 2, 1002, staker3, delegateAmount);
+    await Promise.all([
+      stake(systems(), 2, 1000, staker1, delegateAmount),
```
Comment (noisekit) [inline protocol/synthetix/test/common/stakers.ts:31]: I found earlier that making these things parallel is not working great.

**Comment by davidvuong** [comment]
The broken test for bfp-market seems to be consistent failing. Still investigating the cause but feel free to merge ignoring it

> PerpAccountModule mergeAccounts should revert if called from a non settlement hook

**Review by davidvuong — APPROVED** [review]

**Pull request merged**